tests/test_memory_banks.py

```
@@ -434,9 +442,103 @@ def test_mnemosyne_db_path_override_takes_precedence(self):
+        monkeypatch.setenv("MNEMOSYNE_DATA_DIR", str(data_dir))
+        # Drop the cache so _default_data_dir() re-reads the env var.
+        import mnemosyne.core.banks as banks_mod
+        banks_mod.DEFAULT_DATA_DIR = Path(
+            os.environ.get("MNEMOSYNE_DATA_DIR", banks_mod.DEFAULT_DATA_DIR)
+        )
+        try:
+            assert bank_exists_read_only("gamma") is True
+        finally:
+            # Restore cached default so other tests are unaffected.
+            banks_mod.DEFAULT_DATA_DIR = (
+                Path(os.environ.get("HOME", "/tmp")) / ".hermes" / "mnemosyne" / "data"
```

Ruff (0.15.20)

[error] Probable insecure usage of temporary file or directory: "/tmp"

(S108)